spec/lang/well-formed.md

```diff
@@ -376,13 +425,30 @@ impl PlaceExpr {
                 }
             },
             Deref { operand, ty } => {
+                use SizeStrategy::*;
+
                 let op_ty = operand.check_wf::<T>(locals, prog)?;
-                ensure_wf(matches!(op_ty, Type::Ptr(_)), "PlaceExpr::Deref: invalid type")?;
+                let Type::Ptr(op_ptr_ty) = op_ty else {
+                    throw_ill_formed!("PlaceExpr::Deref: invalid operand type");
+                };
+                let Some(pointee_layout) = op_ptr_ty.pointee() else {
+                    throw_ill_formed!("PlaceExpr::Deref: invalid operand type");
+                };
+                // This breaks some tests, so this is well formed, but just UB?
+                // ensure_wf(ty.size::<T>() == pointee_layout.size, "PlaceExpr::Deref: incompatible return type layout")?;
```

Comment:
Similarly here, deref allows to create a place which isn't the same size as the pointer it dereferences, which actually breaks about 10 tests. I'm not sure if this is really what we want.